### PR TITLE
Feature/kp/product services design

### PR DESCRIPTION
**Pull request by kuldip-trentium**
Screenshot_20240906-115018_CRM
Screenshot_20240906-115021_CRM
Screenshot_20240906-115028_CRM

### CODE REVIEW

File: components/molecules/ActionModal/ActionModal.styles.ts [hunk 1]
```diff
@@ -47,20 +47,20 @@ export const PrimaryLabel = styled(Text)<{ actionType: Actions }>`
     actionType === Actions.success
       ? theme.colors.primaryColor
       : actionType === Actions.delete
-        ? theme.colors.deleteColor
-        : actionType === Actions.default
-          ? theme.colors.white
-          : 'none'};
-  font-size: 14px;
+      ? theme.colors.BlueChaos
```
Comment (Deeptrentium) [inline components/molecules/ActionModal/ActionModal.styles.ts:50]: BlueChaos = blueChaos

File: components/molecules/ActionModal/ActionModal.tsx [hunk 2]
```diff
@@ -54,16 +54,16 @@ const ActionModal: React.FC<RNModalProps> = memo(
...
-          <Spacer size={36} />
+          <Spacer size={16} />
           <ButtonContainer>
             <AllButtonView>
               <CancelPressable onPress={onCancelPress}>
                 <SecondaryLabel>{actiontext}</SecondaryLabel>
               </CancelPressable>
               <ActionPressable onPress={handleSubmit}>
-                {loading && <Loader size={16} color={colors?.deleteColor} />}
+                {loading && <Loader size={16} color={colors?.BlueChaos} />}
```
Comment (Deeptrentium) [inline components/molecules/ActionModal/ActionModal.tsx:66]: BlueChaos blueChaos

File: components/molecules/DocumentPicker/DocumentPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+import { useTranslation } from 'react-i18next';
+import { DocumentPickerProps } from './DocumentPicker.props';
+import { HeaderText } from '@organisms/BasicInformatioForm/BasicInformationForm.styles';
+import { Spacer } from '@atoms/common/common.styles';
+import ActionModal from '@molecules/ActionModal/ActionModal';
+import { deleteLeadDocumentsAction } from '@redux/actions/lead';
+import { useAppDispatch } from '@redux/store';
+import { Actions } from '@molecules/ActionModal/ActionModal.props';
+import { Pressable } from 'react-native';
+import Delete from '@atoms/Illustrations/Delete';
```
Comment (Deeptrentium) [inline components/molecules/DocumentPicker/DocumentPicker.tsx:28]: Delete - DeleteIcon

do this changes every where

File: components/molecules/DocumentPicker/DocumentPicker.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+      const res = await DocumentPicker.getDocumentAsync({
+        type: ['application/pdf', 'image/*', 'text/plain'],
+        copyToCacheDirectory: true,
+      });
+
+      if (!res.canceled) {
+        res.assets.forEach((file) => {
+          if (file.size > MAX_FILE_SIZE) {
+            toast.show(t('fileSizeExceeded'), {
+              type: 'customToast',
```
Comment (Deeptrentium) [inline components/molecules/DocumentPicker/DocumentPicker.tsx:62]: enum

File: components/molecules/DocumentPicker/DocumentPicker.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+  const dispatch = useAppDispatch();
+  const toast = useToast();
+  const { colors } = useAppTheme();
+  const [deleteDocumentUrl, setDeleteDocumentUrl] = useState(null);
+  const [deleteShowModal, setDeleteShowModal] = useState<Boolean>(false);
+  const [deleteLoading, setDeleteLoading] = useState(false);
+  const pickFile = async () => {
+    try {
+      const { status } = await MediaLibrary.requestPermissionsAsync();
+      if (status !== 'granted') {
```
Comment (Deeptrentium) [inline components/molecules/DocumentPicker/DocumentPicker.tsx:50]: add enum

File: components/molecules/DocumentPicker/DocumentPicker.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+    const updatedDocuments = documentArray?.filter(
+      (item) => item?.uri !== deleteDocumentUrl,
+    );
+
+    if (deletedDocument?.[0]?.id) {
+      try {
+        setDeleteLoading(true);
+        const response = await dispatch(
+          deleteLeadDocumentsAction({ media_id: deletedDocument?.[0]?.id }),
```
Comment (Deeptrentium) [inline components/molecules/DocumentPicker/DocumentPicker.tsx:110]: remove ?

File: components/molecules/DocumentPicker/DocumentPicker.tsx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+        ).unwrap();
+        toast.show(response?.message, {
+          type: 'customToast',
```
Comment (Deeptrentium) [inline components/molecules/DocumentPicker/DocumentPicker.tsx:113]: enum

File: components/molecules/DocumentPicker/DocumentPicker.tsx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+          data: {
+            type: ToastTypeProps.Success,
+          },
+        });
+      } catch (error) {
+        toast.show(error, {
+          type: 'customToast',
```
Comment (Deeptrentium) [inline components/molecules/DocumentPicker/DocumentPicker.tsx:120]: enum

File: components/molecules/DocumentPicker/DocumentPicker.tsx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+      )}
+
+      {documentArray?.length > 0 && (
+        <>
+          <Spacer size={8} />
+          <HeaderText>{tb('attachments')}</HeaderText>
+          <Spacer size={16} />
+          <FlatListCon
+            data={documentArray}
+            renderItem={({ item }) => (
```
Comment (Deeptrentium) [inline components/molecules/DocumentPicker/DocumentPicker.tsx:149]: merge both code in one condition

File: components/molecules/LeadDetail/LeadDetail.tsx [hunk 10]
```diff
@@ -1,113 +1,131 @@
...
+      onEdit();
     } else {
-      toast.show(t('canNotFindId'), {
-        type: ToastType.Custom,
-        data: {
-          type: ToastTypeProps.Error,
-        },
-      });
+      if (leadData?.id) {
+        router.navigate(`/(protected)/add-lead/${leadData?.id}`);
```
Comment (Deeptrentium) [inline components/molecules/LeadDetail/LeadDetail.tsx:61]: remove ?

File: components/molecules/LeadDetail/LeadDetail.tsx [hunk 11]
```diff
@@ -1,113 +1,131 @@
...
       </LeadInfoView>
-      {leadData?.phone && (
-        <WhatsAppContainer onPress={handleWhatsApp}>
-          <WhatsApp />
-          <WhatsAppText>{t('whatsapp')}</WhatsAppText>
-        </WhatsAppContainer>
+      {showSocialMedia && (
+        <ContactBox>
+          {leadData?.email && (
+            <WhatsAppContainer onPress={() => handleEmail(leadData?.email)}>
```
Comment (Deeptrentium) [inline components/molecules/LeadDetail/LeadDetail.tsx:114]: remove ?

File: components/molecules/LeadDetail/LeadDetail.tsx [hunk 12]
```diff
@@ -1,113 +1,131 @@
...
+              <EmailSendBox />
+            </WhatsAppContainer>
+          )}
+          {leadData?.phone && (
+            <WhatsAppContainer onPress={() => handlePhoneCall(leadData?.phone)}>
```
Comment (Deeptrentium) [inline components/molecules/LeadDetail/LeadDetail.tsx:119]: WhatsAppContainer = CommunicationOptionCon

File: components/molecules/LeadStatus/LeadStatus.tsx [hunk 13]
```diff
@@ -40,11 +40,7 @@ const LeadStatus: React.FC<LeadStatusProps> = ({ leadStatus }) => {
       );
 
     default:
-      return (
-        <LeadStatusContainer>
-          <LeadStatusText>{t('contacted')}</LeadStatusText>
-        </LeadStatusContainer>
-      );
+      return <></>;
```
Comment (Deeptrentium) [inline components/molecules/LeadStatus/LeadStatus.tsx:43]: null

File: components/molecules/Search/Search.tsx [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+import isEmpty from 'lodash/isEmpty';
+
+const SearchFilter: React.FC<SearchFilterProps> = ({
+  setSearch,
+  search,
+  handleSearch,
+}) => {
+  const { t: ts } = useTranslation('drawer');
+  const { colors } = useAppTheme();
+  const debouncedLeadSearch = useDebounce(search || undefined, 300);
```
Comment (Deeptrentium) [inline components/molecules/Search/Search.tsx:22]: 300 must be come from constant

File: components/molecules/Search/Search.tsx [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+            {
+              backgroundColor: colors.snowflake,
+              borderRadius: 25,
+              overflow: 'hidden',
+              borderColor: colors.primaryColor,
+              paddingLeft: -20,
+              paddingRight: 40,
+            },
+          ]}
+          textColor={colors?.textDark}
```
Comment (Deeptrentium) [inline components/molecules/Search/Search.tsx:52]: remove ?

File: components/molecules/TitleWithButton/TitleWithButton.styles.ts [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+export const BackBtnCon = styled(Pressable)``;
+export const HeaderWithTextContainer = styled(View)<{ top?: number }>`
+  display: flex;
+  flex-direction: row;
+  padding: 0px 16px;
+  padding-bottom: 16px;
+  justify-content: space-between;
+  align-items: center;
+  width: 100%;
+  margin-top: ${({ top }) => top - 20 || 0}px;
```
Comment (Deeptrentium) [inline components/molecules/TitleWithButton/TitleWithButton.styles.ts:14]: update this

File: components/molecules/TitleWithButton/TitleWithButton.styles.ts [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+  justify-content: center;
+`;
+
+export const ButtonView = styled(Pressable)`
+  border-radius: 4px;
+  margin-left: auto;
+  padding-right: 8px;
+`;
+
+export const ButtonText = styled(Text)<{ textColor?: string }>`
```
Comment (Deeptrentium) [inline components/molecules/TitleWithButton/TitleWithButton.styles.ts:55]: add default color

File: components/molecules/TitleWithButton/TitleWithButton.tsx [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+  HeaderWithTextContainer,
+  TextInHeader,
+  IconCon,
+  HeaderTextCon,
+} from './TitleWithButton.styles';
+import { Pressable } from 'react-native';
+import Back from '@atoms/Illustrations/Back';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { router } from 'expo-router';
+interface TitleWithButtonProps {
```
Comment (Deeptrentium) [inline components/molecules/TitleWithButton/TitleWithButton.tsx:13]: move it to props file

File: components/organisms/AddProductForm/AddProductForm.tsx [hunk 19]
```diff
@@ -93,47 +60,16 @@ const AddProductForm: React.FC<AddProductFormProps> = ({
     if (params?.slug) {
       form.change('name', productServiceDetail?.name);
       form.change('description', productServiceDetail?.description);
-      setDocumentArray(productServiceDetail?.documents);
+      console.log(
```
Comment (Deeptrentium) [inline components/organisms/AddProductForm/AddProductForm.tsx:63]: remove log

File: components/organisms/BasicInformatioForm/BasicInformationForm.styles.ts [hunk 20]
```diff
@@ -122,7 +124,7 @@ export const ContainerView = styled(View)`
 export const FormButtonText = styled(Text)<{ valid?: boolean }>`
   font-weight: 700;
   color: ${({ valid, theme }) =>
-    valid ? theme.colors.primaryColor : theme.colors.lightGray};
+    valid ? theme.colors.white : theme.colors.white};
```
Comment (Deeptrentium) [inline components/organisms/BasicInformatioForm/BasicInformationForm.styles.ts:127]: add disable button color once UI updated

File: components/organisms/UserDetailCard/UserDetailCard.tsx [hunk 21]
```diff
@@ -1,70 +1,43 @@
...
-                onDelete();
-              }}
-              title={t('delete')}
-            />
-          </Menu>
-        </ActionMenuContainer>
+        <Spacer size={8} />
+        {data?.email && (
+          <PhoneInfoView>
+            <EmailIcon color={colors?.gray} />
```
Comment (Deeptrentium) [inline components/organisms/UserDetailCard/UserDetailCard.tsx:37]: remove ?

File: components/templates/ScreenTemplate/ScreenTemplate.tsx [hunk 22]
```diff
@@ -11,18 +18,25 @@ const ScreenTemplate: React.FC<ScreenTemplateProps> = ({
...
-        </AddButton>
+      {title && (
+        <TitleWithButton text={title} btnBackPress={() => router?.back()} />
       )}
+      {isDrawerBtn && <MoreMenuButton />}
+      <ScreenTemplateView>
+        {children}
+        {!!addButtonText && (
+          <AddButton onPress={() => onAddButtonPress?.()}>
+            <PlusIcon color={colors?.bgColor} />
```
Comment (Deeptrentium) [inline components/templates/ScreenTemplate/ScreenTemplate.tsx:35]: remove ?

**Review by Deeptrentium — CHANGES_REQUESTED** [review]

File: app/(protected)/(tabs)/dashboard.tsx [hunk 23]
```diff
@@ -144,12 +144,13 @@ const Dashboard = () => {
         key={`${lead?.id}-${index}`}
         onDelete={() => handleDelete(lead?.id)}
         leadData={lead}
+        isShowSocialMedia
```
Comment (sagark1510) [inline app/(protected)/(tabs)/dashboard.tsx:147]: isShowSocialMedia = isSocialMediaVisible

File: app/(protected)/(tabs)/dashboard.tsx [hunk 24]
```diff
@@ -144,12 +144,13 @@ const Dashboard = () => {
...
       />
     </Pressable>
   );
 
   return (
-    <ScreenTemplate>
+    <ScreenTemplate isDrawerBtn>
```
Comment (sagark1510) [inline app/(protected)/(tabs)/dashboard.tsx:153]: isDrawerBtn = moreVisible

File: app/(protected)/(tabs)/leads.tsx [hunk 25]
```diff
@@ -78,10 +77,10 @@ const Leads = () => {
     useState<RefObject<Swipeable> | null>(null);
```
Comment (sagark1510) [inline app/(protected)/(tabs)/leads.tsx:77]: do we still need `Swipeable`?

File: app/(protected)/(tabs)/leads.tsx [hunk 26]
```diff
@@ -78,10 +77,10 @@ const Leads = () => {
...
   const [modal, setModal] = useState(false);
   const [currentId, setCurrentId] = useState<number>(0);
-  const [startDate, setStartDate] = useState('');
-  const [endDate, setEndDate] = useState('');
   const [filterLoading, setFilterLoading] = useState(false);
   const [filterCount, setFilterCount] = useState(0);
```
Comment (sagark1510) [inline app/(protected)/(tabs)/leads.tsx:81]: you don't need to set filter count.. derive it run time.
```
filterCount = useMemo(() => {
 // TODO: add filter count logic
}, [filters]);
```

File: app/(protected)/(tabs)/leads.tsx [hunk 27]
```diff
@@ -78,10 +77,10 @@ const Leads = () => {
...
+  const [startDate, setStartDate] = useState('');
+  const [endDate, setEndDate] = useState('');
   const handleDelete = async (slug: number) => {
     setShowModal(true);
     setDeleteCardId(slug);
```
Comment (sagark1510) [inline app/(protected)/(tabs)/leads.tsx:86]: setDeleteLeadId

File: app/(protected)/(tabs)/products.tsx [hunk 28]
```diff
@@ -2,10 +2,8 @@ import Loader from '@atoms/Loader/Loader';
...
 import { RootState, useAppDispatch, useSelector } from '@redux/store';
 import ScreenTemplate from '@templates/ScreenTemplate/ScreenTemplate';
-import { router, useNavigation } from 'expo-router';
-import React, { RefObject, useEffect, useState } from 'react';
-import { useTranslation } from 'react-i18next';
-import { Swipeable } from 'react-native-gesture-handler';
+import { router } from 'expo-router';
+import React, { useState } from 'react';
 import { useToast } from 'react-native-toast-notifications';
 import { FlatListCon } from './tabs.style';
```
Comment (sagark1510) [inline app/(protected)/(tabs)/products.tsx:8]: FlatListCon ?? rename

File: app/(protected)/(tabs)/products.tsx [hunk 29]
```diff
@@ -16,70 +14,42 @@ import {
...
-    if (openSwipeAbleRef && openSwipeAbleRef.current) {
-      openSwipeAbleRef.current.close();
-    }
-  };
   const [showModal, setShowModal] = useState(false);
-  const [deleteId, setDeleteId] = useState<number | string>(0);
-  const [selectedCard, setSelectedCard] = useState<number | null>(null);
   const [moreLoading, setMoreLoading] = useState(false);
   const [deleteLoading, setDeleteLoading] = useState(false);
+  const [deleteId, setDeleteId] = useState(0);
```
Comment (sagark1510) [inline app/(protected)/(tabs)/products.tsx:28]: deleteId of what?

File: app/(protected)/(tabs)/products.tsx [hunk 30]
```diff
@@ -16,70 +14,42 @@ import {
...
-      />
-    </ProductCardView>
+      onDelete={handleDeleteProduct}
+      onEdit={() => handleEdit(item?.id)}
+      data={item}
+      isServices
+      showModal={showModal}
+      setShowModal={setShowModal}
+      loading={deleteLoading}
+      setDeleteId={setDeleteId}
```
Comment (sagark1510) [inline app/(protected)/(tabs)/products.tsx:51]: never pass state change actions directly.

onChangeDeleteId={(id) => setDeleteId(id)}

File: app/(protected)/(tabs)/users.tsx [hunk 31]
```diff
@@ -38,20 +34,12 @@ const Users = () => {
...
   const userList = useSelector((state: RootState) => state.user?.userList);
-  const [openSwipeAbleRef, setOpenSwipeAbleRef] =
-    useState<RefObject<Swipeable> | null>(null);
-  const [deleteId, setDeleteId] = useState<number | string>(0);
-  const [selectedCard, setSelectedCard] = useState<number | null>(null);
   const [loading, setLoading] = useState(false);
   const [deleteLoading, setDeleteLoading] = useState(false);
+  const [deleteId, setDeleteId] = useState(0);
   const [moreLoading, setMoreLoading] = useState(false);
   const [refreshing, setRefreshing] = useState(false);
```
Comment (sagark1510) [inline app/(protected)/(tabs)/users.tsx:41]: as you can see we have 3 different loading variables we should optimize it.

`const [loadingStatus, setLoadingStatus] = useState<'NONE' | 'DELETE' | 'MORE' | 'REFRESH'>('NONE');`

File: app/(protected)/(tabs)/users.tsx [hunk 32]
```diff
@@ -114,34 +88,26 @@ const Users = () => {
...
-        createdAt={item?.createdAt}
-        closeSwipeAble={closeSwipeAble}
-        setSwipeAbleRef={setSwipeAbleRef}
-        selectedCard={selectedCard}
-        setSelectedCard={setSelectedCard}
-        cardIndex={index}
-        id={0}
-        cardImage={0}
+        data={item}
+        setShowModal={setShowModal}
```
Comment (sagark1510) [inline app/(protected)/(tabs)/users.tsx:104]: remove all set actions from props

File: app/(protected)/(tabs)/users.tsx [hunk 33]
```diff
@@ -114,34 +88,26 @@ const Users = () => {
     }
   };
 
-  const renderTabs = ({
+  const renderUsers = ({
```
Comment (sagark1510) [inline app/(protected)/(tabs)/users.tsx:91]: renderUsers = renderUser

File: components/molecules/ActionModal/ActionModal.styles.ts (hunk 1 above)
Comment (sagark1510) [reply components/molecules/ActionModal/ActionModal.styles.ts:50]: why is this resolved? I still see the changes are not reflected.

File: components/molecules/ActionModal/ActionModal.styles.ts [hunk 34]
```diff
@@ -47,20 +47,20 @@ export const PrimaryLabel = styled(Text)<{ actionType: Actions }>`
...
+      : actionType === Actions.default
```
Comment (sagark1510) [inline components/molecules/ActionModal/ActionModal.styles.ts:51]: No nested ternary

File: lib/utils/common.ts [hunk 35]
```diff
@@ -123,3 +123,14 @@ export const generateWhatsAppUrl = (phoneNumber: number | string) => {
...
 };
+export const handleOpenEmail = (email: string) => {
+  const emailUrl = `mailto:${email}?subject=Your%20Subject%20Here&body=Your%20Message%20Here`;
+
+  Linking.openURL(emailUrl);
+};
+export const handleOpenDialCall = (phoneNumber: number) => {
+  const phoneUrl = `tel:${phoneNumber}`;
+  Linking.openURL(phoneUrl).catch((err) => {
+    console.error('Failed to open dialer', err);
```
Comment (sagark1510) [inline lib/utils/common.ts:134]: inform user about the failure

**Review by sagark1510 — APPROVED** [review]

**Pull request merged**